halpybot/packages/checks/checks.py

```diff
@@ -11,16 +11,19 @@
 import functools
 from typing import List
 from loguru import logger
+from attr import dataclass
 from halpybot import config
 from ..models import User
 from ..database import test_database_connection, NoDatabaseConnection
 
 
+@dataclass
```

Comment:
[semantics] this should probably be immutable at runtime.
```suggestion
@dataclass(frozen=True)
```